contributors/design-proposals/dynamic-admission-control-configuration.md

```diff
@@ -0,0 +1,292 @@
+
+The schema is copied from 
+[#132](https://github.com/kubernetes/community/pull/132) with a few
+modifications.
+
+```golang
+type AdmissionControlConfiguration struct {
+    metav1.TypeMeta 
+
+    // validation will only allow one name in 1.7, it can be relaxed later. 
+    v1.ObjectMeta
+
+    // ResourceInitializers is a list of resources and their default initializers
+    ResourceInitializers []ResourceDefaultInitializer
+
+    ExternalAdmissionHooks []ExternalAdmissionHook
+}
+
+// Because the order of initializers matters, and each resource might need
+// differnt order, the ResourceDefaultInitializers are indexed by Resource.
+type ResourceDefaultInitializer struct {
+    // APIGroup of the resource. Because we want to use APIGroup + Resource as a
+    // merge key, we don't use a Resource struct. See muti-fields merge key
+    // [proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/multi-fields-merge-key.md).
+    APIGroup string
+
+    // Resource identifies the type of resource to be initialized that should be
+    // initialized
+    Resource string
+
+    // Initializers are the default names that will be registered to this resource
+    Initializers []Initializer
+}
+
+type Initializer struct {
+    // Name is the string that will be registered to the resource that needs
+    // initialization.
+    Name string
+
+    // FailurePolicy defines what happens if there is no initializer controller
+    // takes action. Allowed values are Ignore, or Fail. If "Ignore" is set, 
+    // apiserver removes initilizer from the initializers list of the resource
+    // if the timeout (30s) is reached; If "Fail" is set, apiserver returns 
+    // timeout error if the timeout is reached.
+    FailurePolicy FailurePolicyType
+}
+
+type FailurePolicyType string
+
+const (
+    Ignore FailurePolicyType = "Ignore"
+    // **optional** For 1.7, only "Ignore" is allowed. We can add "Fail" when
+    // the feature is more mature.
+    Fail FailurePolicyType = "Fail"
+)
+
+type ExternalAdmissionHook struct {
+    // Name of the AdmissionHook. It must be unique. It is used as the merge key.
+    Name string
+
+    // ClientConfig defines how to talk to the hook.
+    ClientConfig AdmissionHookClientConfig
+
+    // Rules describes what operations on what resources/subresources the webhook cares about.
+    // The webhook cares about an operation if it matches any Rule.
+    // Each rule must has a unique APIGroup.
+    Rules []Rule
+
+    // FailurePolicy defines how unrecognized errors from the admission endpoint are handled -
+    // allowed values are Ignore, Fail. Default value is Fail
+    FailurePolicy FailurePolicyType
+}
+
+type Rule struct {
+    // Verbs is the list of verbs this hook will be invoked on - POST, PUT, or *
+    // for all operations. Defaults to '*'.
+    Verbs []OperationType
+
+    // APIGroup is the API group the resources belong to. '*' is all groups.
+    APIGroup string 
```

Comment:
@lavalamp @smarterclayton @whitlockjc I have two concerns with rbac's schema,
* Different `Rules` can have the same effect, i.e., hard for a human to tell if two sets of Rules are the same;
* Given a resource, it's hard for a human to locate the rules that are applied to it

So I introduced one change to ExternalAdmissionHook's `Rules`, i made the APIGroup a string instead of a slice, and restrict that each `Rule` in `Rules` must have a different `APIGroup`. This solves the two concerns i had. @whitlockjc because the uniqueness is enforced at the APIGroup level, it incurs the minimal amount of duplication.